### PR TITLE
MGDAPI-3855 - Implement persistent volume for alert manager

### PR DESCRIPTION
**Pull request by valerymo**
Implement persistent volume for alert manager
Jira: https://issues.redhat.com/browse/MGDAPI-3855
Integrately PR: https://github.com/integr8ly/integreatly-operator/pull/2646

TEST
1. Check that Alertmanager PVC created in `redhat-rhoam-observability` namespace. For example:
$ oc get pvc -n redhat-rhoam-observability
NAME                                                       STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   AGE
alertmanager-alertmanager-db-alertmanager-alertmanager-0   Bound    pvc-60903afd-b8b0-43c1-8fc4-1adc4062fe01   1Mi        RWO            gp2            3h2m
prometheus-prometheus-db-prometheus-prometheus-0           Bound    pvc-4ae289f2-5fa8-42f1-9864-19aa805e0760   50Gi       RWO            gp2            3h2m
2. Silent/Snooze any alert (for example alertname="DeadMansSwitch"), using Alertmanager (or any other way), kill alertmanager pod , and see that alert still snoozed

### CODE REVIEW

File: PROJECT [hunk 1]
```diff
@@ -3,9 +3,15 @@ layout: go.kubebuilder.io/v2
...
 resources:
-- group: observability
+-
+  # TODO(user): Uncomment the below line if this resource implements a controller, else delete it.
+  # controller: true
+  domain: redhat.com
+  group: observability
   kind: Observability
+  # TODO(user): Update the package path for your API if the below value is incorrect.
+  path: github.com/redhat-developer/observability-operator/v3/api/v1
```
Comment (makslion) [inline PROJECT:13]: Is this PR is till "WIP" and should these TODOs be acted on? :thinking:

Comment (valerymo) [reply PROJECT:13]: Thank you very much for comment. Subject updated now (forgot : ). Regarding `PROJECT` file - it was generated, as changes required moving from version `3-alpha` to `3`. I will try recheck it with OO people before removing `TODO` comments. Thanks again for comment @makslion

Comment (valerymo) [reply PROJECT]: @makslion Thank you for comments. Update done. OO bundles created now with operator-sdk 1.4.2 (as defined in OO in observability-operator.clusterserviceversion.yaml, annotations.yaml). PROJECT file (and few  others) restored as in main branch

**Comment by makslion** [comment]
the change looks good. Was unable to install sole OO to the cluster but alongside RHOAM it behaved well. This PR still would need approval from the OO team and this allows me to drop an

/lgtm

**Comment by pb82** [comment]
@vmanley can you please take a look when you get a chance?

**Comment by vmanley** [comment]
@valerymo is there a plan to have an override for alert manager storage from the observability resources repo? Currently there's no `overrideAlertmanagerPvcSize` field in the resources repo

**Comment by vmanley** [comment]
Change works as described. Default 1Gi PVC created by Operator for Alert Manager and verified by silencing alert. If there is no plan to introduce an override from observability-resources-mk repo then the check for alertmanagerConfig.OverrideAlertmanagerPvcSize won't be relevant. @pb82 what do you think about keeping check in case we want to implement an override in future?

**Comment by vmanley** [comment]
@valerymo going to keep override but no need for your PR on github.com/bf2fc6cc711aee1a0c2a/observability-resources-mk/pull/205 above. You can delete that WIP PR. Thanks for your work on this and sorry again for the delay

**Pull request merged**